test/js/array_join.spec.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+ **/
+const { TaintedUtils, taintFormattedString, formatTaintedValue } = require('./util')
+const assert = require('assert')
+
+describe('String case operator', function () {
+  const id = TaintedUtils.createTransaction('1')
+
+  const rangesTestCases = [
+    {
+      testArray: [':+-a-+:'],
+      testSeparator: undefined,
+      joinResult: ':+-a-+:'
+    },
+    {
+      testArray: [':+-a-+:'],
+      testSeparator: '###',
+      joinResult: ':+-a-+:'
+    },
+    {
+      testArray: [':+-a-+:'],
+      testSeparator: 123,
+      joinResult: ':+-a-+:'
+    },
+    {
+      testArray: [':+-a-+:'],
+      testSeparator: ':+-###-+:',
+      joinResult: ':+-a-+:'
+    },
+    {
+      testArray: [':+-abc-+:', ':+-def-+:'],
+      testSeparator: undefined,
+      joinResult: ':+-abc-+:,:+-def-+:'
+    },
+    {
+      testArray: [':+-abc-+:', ':+-def-+:'],
+      testSeparator: '###',
+      joinResult: ':+-abc-+:###:+-def-+:'
+    },
+    {
+      testArray: [':+-abc-+:', ':+-def-+:'],
+      testSeparator: 123,
+      joinResult: ':+-abc-+:123:+-def-+:'
+    },
+    {
+      testArray: [':+-abc-+:', 'def'],
+      testSeparator: undefined,
+      joinResult: ':+-abc-+:,def'
+    },
+    {
+      testArray: [':+-abc-+:', 'def'],
+      testSeparator: '###',
+      joinResult: ':+-abc-+:###def'
+    },
+    {
+      testArray: [':+-abc-+:', 'def'],
+      testSeparator: 123,
+      joinResult: ':+-abc-+:123def'
+    },
+    {
+      testArray: ['abc', 'def'],
+      testSeparator: ':+-###-+:',
+      joinResult: 'abc:+-###-+:def'
+    },
+    {
+      testArray: [':+-abc-+:', 'def', 'gh:+-ij-+:kl'],
+      testSeparator: ':+-###-+:',
+      joinResult: ':+-abc-+::+-###-+:def:+-###-+:gh:+-ij-+:kl'
+    },
+    {
+      testArray: [':+-abc-+:', 'def', 'gh:+-ij-+:kl'],
+      testSeparator: 'AB:+-CD-+:EF',
+      joinResult: ':+-abc-+:AB:+-CD-+:EFdefAB:+-CD-+:EFgh:+-ij-+:kl'
+    },
+    {
+      testArray: [':+-abc-+:', 'def', 'gh:+-ij-+:kl'],
```

Comment:
what if `testArray` contains mixed string and non-string types ie `testArray: [':+-abc-+:', 666, {a:1}]`

Reply:
For this case, object is casted to string `[object Object]` in js `Array.join`. 

> [':+-abc-+:', 666, {a:1}].join()
> ':+-abc-+:,666,[object Object]'

We are doing the same, since the length used to compute the offset for the result ranges, is the one of the string representation of the object. See https://github.com/DataDog/dd-native-iast-taint-tracking-js/blob/main/src/utils/string_utils.h#L63

Same happens for `undefined` and null `values`.

I will add additional test cases for these values.